e2e/specs/results-viewer.spec.ts

```diff
@@ -0,0 +1,130 @@
+    await page.locator('#ManualEntryPlateletsid').fill('10');
+    await page.locator('#ManualEntryNeutrophilsMicroscopicid').fill('10');
+    await page.locator('#ManualEntryMCVid').fill('10');
+    await page.locator('#ManualEntryMCHid').fill('10');
+    await page.locator('#ManualEntryMCHCid').fill('10');
+    await page.locator('#ManualEntryLymphocytesMicroscopicid').fill('10');
+    await page.locator('#ManualEntryHematocritid').fill('10');
+    await page.locator('#ManualEntryHaemoglobinid').fill('10');
+    await page.locator('#ManualEntryCombinedPercentageMonocytesEosinophilsBasophilsid').fill('10');
+  });
+
+  await test.step('When I click the `Chemistry` tab', async () => {
+    await page.getByRole('button', { name: 'Chemistry', exact: true }).click();
+  });
+
+  await test.step('And Then I fill the chemistry results form', async () => {
+    await page.locator('#ManualEntryAlkalinePhosphataseid').fill('10');
+    await page.locator('#ManualEntryAmylaseid').fill('10');
+    await page.locator('#ManualEntryBUNid').fill('10');
+    await page.locator('#manualEntryFastingBloodGlucosemgdlid').fill('10');
+    await page.locator('#manualEntryPostPrandialBloodGlucosemgdlid').fill('10');
+    await page.locator('#manualEntrySerumAlbuminid').fill('10');
+    await page.locator('#manualEntrySerumCalciumid').fill('10');
+    await page.locator('#manualEntrySerumPotassiumid').fill('10');
+    await page.locator('#manualEntrySerumSodiumid').fill('10');
+    await page.locator('#manualEntrySerumCreatinineid').fill('10');
+    await page.locator('#manualEntryTotalProteinid').fill('10');
+    await page.locator('#manualEntrySerumGlucosemgdlid').fill('10');
+    await page.locator('#manualEntrySerumGlucosemmolid').fill('10');
+    await page.locator('#manualEntryTotalBilirubinid').fill('10');
+    await page.locator('#manualEntrySerumGlutamicOxaloaceticTransaminaseid').fill('10');
+    await page.locator('#manualEntryAlkalinePhosphastaseid').fill('10');
+    await page.locator('#manualEntrySerumUricAcidid').fill('10');
+    await page.locator('#manualEntryTotalCholesterolid').fill('10');
+    await page.locator('#manualEntryTriglyceridesid').fill('10');
+    await page.locator('#manualEntryAmylaseid').fill('10');
+    await page.locator('#manualEntrySerumCarbonDioxideid').fill('10');
+  });
+
+  await test.step('When I click the `Other` tab', async () => {
+    await page.getByRole('button', { name: 'Other', exact: true }).click();
+  });
+
+  await test.step('And Then I fill the other test results', async () => {
+    await page.locator('#manualEntryCultureandSensitivityUrineid').fill('83%');
+    await page.locator('#manualEntryUrinePregnancyTestid_1').check();
+    await page.locator('#manualEntryUrineProteinDipStickid_0').check();
+    await page.locator('#manualEntryUrineBacteriuriaTestid_1').check();
+    await page.locator('#manualEntryErythrocytesPresenceInUrineSedimentByLightMicroscopyTestid_1').check();
```

Comment:
is there any other way to locate elements instead of the id?

Reply:
@jayasanka-sack I understand we want to avoid get by id as much as possible, but in this case, this is the only safe way. we can't do by text or label because a lot of options and values are repeated across questions. and some how, labels don't seem to link to the form inputs for input types.

Reply:
awesome!